lighthouse/src/main.rs

```diff
@@ -619,21 +616,50 @@ fn run<E: EthSpec>(
         environment_builder,
     );
 
-    let logging = tracing_subscriber::registry()
-        .with(dependency_log_filter)
-        .with(file_logging_layer.with_filter(logger_config.logfile_debug_level))
-        .with(stdout_logging_layer.with_filter(logger_config.debug_level))
-        .with(MetricsLayer)
-        .with(libp2p_discv5_layer);
+    let workspace_filter = build_workspace_filter()?;
 
-    let logging_result = if let Some(sse_logging_layer) = sse_logging_layer_opt {
-        logging.with(sse_logging_layer).try_init()
-    } else {
-        logging.try_init()
-    };
+    let mut logging_layers = Vec::new();
+
+    logging_layers.push(
+        stdout_logging_layer
+            .with_filter(logger_config.debug_level)
+            .with_filter(workspace_filter.clone())
+            .boxed(),
+    );
+
+    if let Some(file_logging_layer) = file_logging_layer {
+        logging_layers.push(
+            file_logging_layer
+                .with_filter(logger_config.logfile_debug_level)
+                .with_filter(workspace_filter)
+                .boxed(),
+        );
+    }
+
+    if let Some(sse_logging_layer) = sse_logging_layer_opt {
+        logging_layers.push(sse_logging_layer.boxed());
+    }
+
+    if let Some(libp2p_discv5_layer) = libp2p_discv5_layer {
+        logging_layers.push(
+            libp2p_discv5_layer
+                .with_filter(
+                    EnvFilter::builder()
+                        .with_default_directive(Level::DEBUG.into())
+                        .from_env_lossy(),
+                )
```

Comment:
I also noticed we lost the ability to control the log level with `RUST_LOG` so I've restored it here. It will now default to `debug` and then be overridden by `RUST_LOG`